sound/soc/sof/intel/cnl.c

```diff
@@ -150,14 +151,53 @@ static void cnl_ipc_dsp_done(struct snd_sof_dev *sdev)
 				CNL_DSP_REG_HIPCCTL_DONE);
 }
 
+static bool cnl_is_ipc_compact(struct snd_sof_ipc_msg *msg)
+{
+	/* the compact IPC subset might be extended later */
+	if (msg->header == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE))
+		return true;
+	else
+		return false;
+}
+
+static void cnl_compact_ipc_compress(struct snd_sof_ipc_msg *msg,
+				     u32 *dr, u32 *dd)
+{
+	struct sof_ipc_pm_gate *pm_gate;
+
+	if (msg->header == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE)) {
```

Comment:
why do you need this test here, you're already determined it's a compact IPC so why retest? 
Put differently, it's an error if you call this function with a regular IPC

Reply:
@plbossart @lyakh the definition of the functions cnl_is_ipc_compact() and cnl_compact_ipc_compress() is not for PM_GATE IPC only, it can be extended to e.g. PM_CTX_SAVE/RESTORE, so in cnl_is_ipc_compact() we only check if the on-going IPC is in this sebset, and will do the specific handling in cnl_compact_ipc_compress() for each compact IPC.

I admit we can have better handling to this, e.g. pre-define an array for the subset, as there was this only one IPC, I just used simple "==" check here.

Let me extend this now.

Reply:
The difficulty of using switch just explained in another reply to Pierre.